parachain/pallets/basic-channel/src/outbound/mod.rs

```diff
@@ -164,11 +164,6 @@ pub mod pallet {
 			let principal = Self::principal();
 			ensure!(principal.is_some(), Error::<T>::NotAuthorized,);
 			ensure!(*who == principal.unwrap(), Error::<T>::NotAuthorized,);
-			ensure!(
-				<MessageQueue<T>>::decode_len().unwrap_or(0) <
-					T::MaxMessagesPerCommit::get() as usize,
```

Comment:
With the queue being a BoundedVec, we don't need this validation anymore do we?

Reply:
That is correct, `try_mutate` will only mutate if `Ok()` is returned. So it is safe to drop this `ensure!`.

https://docs.substrate.io/rustdocs/latest/src/frame_support/storage/mod.rs.html#178-179